development/tools/cmd/secretspopulator/main.go

```diff
@@ -0,0 +1,207 @@
+	panicOnError(err)
+
+	kmsService, err := cloudkms.New(client)
+	panicOnError(err)
+
+	p := SecretsPopulator{
+		secretsClient: secretInterface,
+		storageClient: storageClient,
+		kmsClient:     kmsService,
+	}
+
+	secrets, err := readSecretDefFile(*argSecretsDefFile)
+	panicOnError(err)
+
+	panicOnError(p.PopulateSecrets(ctx, *argProject, secrets, *argBucket, *argKeyring, *argKey, *argLocation))
+
+}
+
+// SecretsPopulator is responsible for populating secrets
+type SecretsPopulator struct {
+	secretsClient typedv1.SecretInterface
+	storageClient *storage.Client
+	kmsClient     *cloudkms.Service
+}
+
+// PopulateSecrets populates secrets
+func (s *SecretsPopulator) PopulateSecrets(ctx context.Context, project string, secrets []SecretModel, bucket, keyring, key, location string) error {
+	parentName := fmt.Sprintf("projects/%s/locations/%s/keyRings/%s/cryptoKeys/%s",
+		project, location, keyring, key)
+
+	for _, sec := range secrets {
+		name := fmt.Sprintf("%s.%s", sec.Prefix, fileNameExtension)
+		objHandle := s.storageClient.Bucket(bucket).Object(name)
+		fmt.Printf("Reading object [%s] from bucket [%s]\n", name, bucket)
+		r, err := objHandle.NewReader(ctx)
+		if err != nil {
+			return errors.Wrapf(err, "while reading object [%s] from bucket [%s]", name, bucket)
+		}
+		buf := &bytes.Buffer{}
+
+		if _, err := io.Copy(buf, r); err != nil {
+			return errors.Wrapf(err, "while coping file [%s] to buffer", name)
+		}
+
+		decryptCall := s.kmsClient.Projects.Locations.KeyRings.CryptoKeys.Decrypt(parentName, &cloudkms.DecryptRequest{
+			Ciphertext: base64.StdEncoding.EncodeToString(buf.Bytes()),
+		})
+
+		resp, err := decryptCall.Do()
+		if err != nil {
+			return errors.Wrap(err, "while making decrypt call")
+		}
+
+		decoded, err := base64.StdEncoding.DecodeString(string(resp.Plaintext))
+		if err != nil {
+			return err
+		}
+
+		curr, err := s.secretsClient.Get(sec.Prefix, metav1.GetOptions{})
+		switch {
+		case err == nil:
+			fmt.Printf("Updating k8s secret [%s]\n", curr.Name)
+			curr.Data[sec.Key] = decoded
+			if _, err = s.secretsClient.Update(curr); err != nil {
+				return errors.Wrap(err, "while updating secret")
+			}
+
+		case k8serrors.IsNotFound(err):
+			fmt.Printf("Creating k8s secret [%s] with key [%s] \n", sec.Prefix, sec.Key)
+			if _, err := s.secretsClient.Create(&corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: sec.Prefix,
+				},
+				Data: map[string][]byte{
+					sec.Key: decoded,
+				},
+			}); err != nil {
+				return errors.Wrapf(err, "while creating secret [%s]", sec.Prefix)
+			}
+		default:
+			return errors.Wrapf(err, "while getting secret [%s]", sec.Prefix)
+		}
+
+	}
+	return nil
+}
+
+// RequiredSecretsData represents secrets required by Prow cluster
+type RequiredSecretsData struct {
+	ServiceAccounts []SASecret
+	Generics        []GenericSecret
+}
+
+// SASecret represents Service Account Secret
+type SASecret struct {
```

Comment:
Do we need to handle SA differently than other secrets?

Reply:
In case of service account, secret key will be 'service-account.json' - separate type is introduced to avoid repetition in yaml file.